src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs

```diff
@@ -356,25 +362,81 @@ private protected override TModel TrainModelCore(TrainContext context)
             InitializeBeforeTraining();
 
             Host.CheckValue(context, nameof(context));
-
             Dataset dtrain = null;
             Dataset dvalid = null;
-            CategoricalMetaData catMetaData;
+
             try
             {
-                using (var ch = Host.Start("Loading data for LightGBM"))
+                if (LightGbmTrainerOptions.LightGbmModel != null)
                 {
-                    using (var pch = Host.StartProgressChannel("Loading data for LightGBM"))
+                    using (var ch = Host.Start("Loading LightGBM model file"))
                     {
-                        dtrain = LoadTrainingData(ch, context.TrainingSet, out catMetaData);
-                        if (context.ValidationSet != null)
-                            dvalid = LoadValidationData(ch, dtrain, context.ValidationSet, catMetaData);
+                        StreamReader reader = new StreamReader(LightGbmTrainerOptions.LightGbmModel);
+                        string modelText = reader.ReadToEnd();
+
+                        AdditionalLoadPreTrainedModel(modelText);
+
+                        var modelParameters = Booster.GetParameters(modelText);
+                        // Going to set the parameters via reflection so that we don't have manually set them on the options object
+                        Type optionsType = LightGbmTrainerOptions.GetType();
```

Comment:
@tannergooding there is where I could use your eyes. Basically I am reading in values from a text file and need to store them in their correct places in a C# object. Using reflection so I don't have to hardcode everything since there are lots of values. Is there a better way to do it? Anything wrong with what I am doing?